### ISSUE TITLE
Handling of "bad" packets.

### ISSUE THREAD

**Issue by daid** [jstasiak/python-zeroconf#17]
I was testing some stuff with this zeroconf library on our local network at the office. And I noticed a lot of error messages like this:

```
ERROR:zeroconf:Unknown error, possibly benign: IndexError('index out of range',)
Traceback (most recent call last):
  File "/home/ultimaker/zeroconf.py", line 832, in run
    self.readers[socket_].handle_read(socket_)
  File "/home/ultimaker/zeroconf.py", line 886, in handle_read
    msg = DNSIncoming(data)
  File "/home/ultimaker/zeroconf.py", line 464, in __init__
    self.read_questions()
  File "/home/ultimaker/zeroconf.py", line 481, in read_questions
    name = self.read_name()
  File "/home/ultimaker/zeroconf.py", line 560, in read_name
    length = indexbytes(self.data, off)
IndexError: index out of range
```

I quickly captured the data with wireshark, and noticed I was getting broadcast UDP packages on port 5353 with the contents:
```
{"command","broadcast"}
```
Yes, exactly that text contents. No binary MDNS encoding around it.

The packets seem to origin from an AppleTV device. No idea why they wouldn't be using normal MDNS packets. But it's common enough to say that the library needs some extra hardening on the parsing area.

I can make a patch, but what would be the preferred way to handle this?
(I also have a bugfix for windows, but I'll post that later when I'm 100% sure about the solution)

**Comment by jstasiak** [comment]
@daid Hey, thanks for the feedback. I agree, exception handling is one of the things that can be improved in this code. I'd leave https://github.com/jstasiak/python-zeroconf/blob/b5fb3e86a688f6161c1292ccdffeec9f455c1fbd/zeroconf.py#L833 not to break too much but I think I'd
1. make ``DNSIncoming`` raise some custom ``BadPacket`` error
2. make this code handle it silently (maybe log some debugging information? I'm not a fan of hiding issues but it's either that or letting some malicious/dumb service on the network flood your logs with stack traces)

PS. Any Windows-related patches will be appreciated.

**Comment by daid** [comment]
I think logging it a debug level would be good. As these are UDP broadcasts, you can kinda expect bad packets.

**Comment by stephenrauch** [comment]
Closed via #68
